### PR TITLE
Introduce a helper ExpressionEvaluator to simplify expression evaluation

### LINKED ISSUE #5105: Introduce a helper to evaluate scalar expression
So far, we don't have a effective way to evaluate scalar expressions into `ColumnRef`.

The `ExpressionExecutor` is too complicated for normal use.

### PR DESCRIPTION
**Pull request by leiysky**
Signed-off-by: leiysky <leiysky@outlook.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Introduce a helper `ExpressionEvaluator` which evaluates scalar expression in a recursive fashion.

Compare to `ExpressionExecutor`, `ExpressionEvaluator` doesn't rely on `DataSchema`, and won't produce intermidiate result. It's useful when you want to evaluate an expression on the fly.

## Changelog

- New Feature

## Related Issues

Fixes #5105

### CODE REVIEW

File: query/src/common/expression_evaluator.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+use crate::sessions::QueryContext;
+
+pub struct ExpressionEvaluator;
+
+impl ExpressionEvaluator {
+    pub fn eval(
+        expression: &Expression,
+        block: &DataBlock,
+        ctx: &QueryContext,
```
Comment (BohuTANG) [inline query/src/common/expression_evaluator.rs:36]: 'ctx' is generally used as the first parameter of the function

File: query/src/common/expression_evaluator.rs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    ) -> Result<ColumnRef> {
+        let func_ctx = FunctionContext {
```
Comment (sundy-li) [inline query/src/common/expression_evaluator.rs:38]: Better to add `fn get_function_ctx` inside `ctx`, thus we no longer need ctx.

File: query/src/common/expression_evaluator.rs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            tz: String::from_utf8(ctx.get_settings().get_timezone()?).map_err(|_| {
+                ErrorCode::LogicalError("Timezone has been checked and should be valid.")
+            })?,
+        };
+        match expression {
+            Expression::Column(name) => block.try_column_by_name(name.as_str()).cloned(),
+
+            Expression::Alias(_, _) => Err(ErrorCode::LogicalError(
```
Comment (zhang2014) [inline query/src/common/expression_evaluator.rs:46]: ErrorCode::Unimplemented is better.

File: query/src/common/expression_evaluator.rs (hunk 1 above)
Comment (leiysky) [reply query/src/common/expression_evaluator.rs]: Fixed.

File: query/src/common/expression_evaluator.rs (hunk 2 above)
Comment (leiysky) [reply query/src/common/expression_evaluator.rs]: Fixed.

File: query/src/common/expression_evaluator.rs (hunk 3 above)
Comment (leiysky) [reply query/src/common/expression_evaluator.rs]: Fixed.

**Review by sundy-li — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**